src/Node/ClassStatementsGatherer.php

```diff
@@ -66,11 +73,11 @@ public function getProperties(): array
 	}
 
 	/**
-	 * @return ClassPropertyNode[]
+	 * @return array<int, int>
 	 */
-	public function getTraitProperties(): array
+	public function getIsTraitProperty(): array
```

Comment:
I don't understand this API at all :) What's `array<int, int>`?

It'd be much nicer to add this information `ClassPropertyNode` - a new `isDeclaredInTrait` method.

Reply:
agreed, it's a bit weird to use / understand. this is basically what `isOptionalProperty` is in `ConstantArrayType`, so it works with the numeric index. Such a method would be way nicer though, agreed. Will take a look at it in the evening.